src/NuGet.Core/NuGet.Protocol/HttpSource/HttpRetryHandler.cs

```diff
@@ -209,7 +209,7 @@ internal HttpRetryHandler(IEnvironmentVariableReader environmentVariableReader)
                             success = false;
                         }
                     }
-                    catch (OperationCanceledException)
+                    catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
```

Comment:
`OperationCanceledException` is not listed under `Exceptions` section in [HttpClient.SendAsync](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.sendasync?view=netframework-4.7.2) docs.
Should we use [TaskCanceledException](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcanceledexception?view=netframework-4.7.2) instead of adding a condition.

Reply:
From [the docs on task cancellation](https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/task-cancellation):

> If you are waiting on a Task that transitions to the Canceled state, a [System.Threading.Tasks.TaskCanceledException](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcanceledexception) exception (wrapped in an [AggregateException](https://docs.microsoft.com/en-us/dotnet/api/system.aggregateexception) exception) is thrown.

Therefore, if the cancellation token requests cancellation, this exception handler might see a `TaskCanceledException`, rather than `CancelationToken.ThrowIfCancelationRequested()`'s `OperationCancelledException`.

Therefore, the filter would still be required.

Reply:
Thank you for the details. After reading https://github.com/dotnet/runtime/discussions/49116 and then browsing code in runtime repo, I learnt that `TaskCanceledException` inherits from `OperationCanceledException`.
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskCanceledException.cs#L21

Given this information, I am comfortable with this PR.